Towny/src/main/java/com/palmergames/bukkit/towny/db/TownyDatabaseHandler.java

```diff
@@ -256,6 +256,9 @@ public void removeResident(Resident resident) {
 
 		// Remove resident from towns' outlaw & trusted lists.
 		for (Town town : universe.getTowns()) {
+			if (!universe.hasTown(town.getName()))
```

Comment:
Since this method is used a few times across the PR it might be nice to make a new TownyObject#isRegistered method, annotated with ApiStatus.Internal of course